dataclients/kubernetes/kube.go

```diff
@@ -1213,6 +1213,15 @@ func computeBackendWeights(backendWeights map[string]float64, rule *rule) {
 				// give subsequent backends a higher relative
 				// weight.
 				sc.sum -= weight
+
+				// noops are required to make sure that routes are in order selected by routing tree
+				noopCount := sc.weightsCount - 2
```

Comment:
Not sure what the 2 is here? I guess this is related to my comment above about only counting positive weights and wouldn't be needed in that case?

Reply:
Basically we only need a Noop() for >2 traffic switching routes, because the last  has no Traffic() and the other one has 1 Traffic(), so 1 predicate more than the one without Traffic().
For the >1 with Traffic() routes we need to be sure we have a decreasing number of matching predicates.

Reply:
Then I think a check like `sc.weightsCount > 2` would be simpler to understand in the code.

Reply:
I guess done :)